pkg/controller/service/normalizing.go

```diff
@@ -71,3 +72,68 @@ func matchesKey(obj unstructured.Unstructured, key normalizerKey) bool {
 	}
 	return true
 }
+
+// flexEqual compares two values with special handling for numeric types.
+// This is necessary because JSON unmarshalling (used in Unstructured objects)
+// treats all numbers as float64 by default, while Kubernetes API types (like int32/int64)
+// retain their original integer type when fetched using controller-runtime's client.
+//
+// For example:
+// - A value like `replicas: 1` from YAML becomes float64(1.0)
+// - The same field from the Kubernetes API may be int64(1) or int32(1)
+// - reflect.DeepEqual would return false even though they represent the same number
+//
+// This function ensures such semantically equal numeric values are treated as equal.
+func flexEqual(a, b interface{}) bool {
+	// Compare numeric types with type coercion
+	if isNumber(a) && isNumber(b) {
+		af := toFloat64(a)
+		bf := toFloat64(b)
+		return math.Abs(af-bf) < 1e-9 // tolerate float precision errors
+	}
+
+	// Fall back to deep equal
+	return reflect.DeepEqual(a, b)
+}
+
+func isNumber(v interface{}) bool {
+	switch v.(type) {
+	case int, int8, int16, int32, int64,
+		uint, uint8, uint16, uint32, uint64,
+		float32, float64:
+		return true
+	default:
+		return false
+	}
+}
+
+func toFloat64(v interface{}) float64 {
+	switch n := v.(type) {
+	case int:
+		return float64(n)
+	case int8:
+		return float64(n)
+	case int16:
+		return float64(n)
+	case int32:
+		return float64(n)
+	case int64:
+		return float64(n)
+	case uint:
+		return float64(n)
+	case uint8:
+		return float64(n)
+	case uint16:
+		return float64(n)
+	case uint32:
+		return float64(n)
+	case uint64:
+		return float64(n)
+	case float32:
+		return float64(n)
+	case float64:
```

Comment:
You can use the same case as in `isNumber`:

```go
case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64:
```

Reply:
Unfortunately, I can't do this:
<img width="863" height="541" alt="image" src="https://github.com/user-attachments/assets/afb37956-8595-43a1-8cb9-829f8b3ee07c" />